python/kfserving/kfserving/model.py

```diff
@@ -1,9 +1,11 @@
 class KFModel(object):
+
+    def __init__(self, name: str):
+        self.name = name
+        self.ready = False
+
     def predict(self, inputs):
-            raise NotImplementedError
-    
-    def preprocess(self, inputs):
```

Comment:
I actually think we should keep these interfaces in. Eventually I see this class having 

load
predict
preprocess
postprocess
explain
detectOutlier

etc. 

That allows us the flexibility of a microservices approach or a singleton (all running in one server, may be nice for local testing), Thinking on it, maybe it's best to leave it out for now, but that's what I'm thinking.

Reply:
> That allows us the flexibility of a microservices approach or a singleton (all running in one server, may be nice for local testing), Thinking on it, maybe it's best to leave it out for now, but that's what I'm thinking.

So if I implement a predict and an explain and select the tensorflow protocol what will the request and response look like? 

* Do I get to select whether the explainer runs in the request?
* How will I find the explanation in the response?

Reply:
> > That allows us the flexibility of a microservices approach or a singleton (all running in one server, may be nice for local testing), Thinking on it, maybe it's best to leave it out for now, but that's what I'm thinking.
> 
> So if I implement a predict and an explain and select the tensorflow protocol what will the request and response look like?
> 
> * Do I get to select whether the explainer runs in the request?

TBD

> * How will I find the explanation in the response?

TBD


;)